### DISCUSSION TITLE
An error occurred: giscus is not installed on this repository

### DISCUSSION THREAD

**Discussion by syeda-git** [giscus/giscus#1161] [Q&A]
Hello,

I have installed giscus on the repository, but the giscus app shows that it's not installed on the repository. Here's the error:

![image](https://github.com/giscus/giscus/assets/44337085/fbc5753c-0ba8-43c1-b113-3ae1b772a986)

Here's the snapshot of the installed app on the repo. 

![image](https://github.com/giscus/giscus/assets/44337085/c484ade7-626c-489f-b644-14444d2ca6c5)

Select Repo

![image](https://github.com/giscus/giscus/assets/44337085/9fddd858-5882-4156-a898-623363d88d45)

Syed

**Comment by laymonage** [comment]
You probably haven't set the correct details for the app e.g. the `GITHUB_APP_ID`, `GITHUB_CLIENT_ID`, or the `GITHUB_CLIENT_SECRET`.

Try changing this line:

https://github.com/giscus/giscus/blob/d1a67e8b239de6195171bf3ef8cef207e3b8f3a6/services/github/getAppAccessToken.ts#L27

to:

```diff
-  }).then((response) => response.json());
+  }).then((response) => response.json()).then((data) => {console.log(data); return data;});
```

and check your logs to see GitHub's response.

**Comment by syeda-git** [comment]
When i start the server, these messages are printed on the console. Let me know if i have to check elsewhere for the logs.

![image](https://github.com/giscus/giscus/assets/44337085/04c4ab6a-c520-4297-ab47-34caabc67819)

Here's the env file. The App ID, Client ID, and Secret seems to be correct.




App settings:

![image](https://github.com/giscus/giscus/assets/44337085/1a491ff8-04d5-47cc-96fb-46ea3ac55fb8)

Syed

**Reply by laymonage** [reply]
Are you using GitHub Enterprise by any chance? The App ID seems low for a newly-created app on GitHub.com. If you do, it's not officially supported yet, see #1141. You'll need to change the code to point to your GHES installation, e.g. here:

https://github.com/giscus/giscus/blob/d1a67e8b239de6195171bf3ef8cef207e3b8f3a6/services/config.ts#L3

here:

https://github.com/giscus/giscus/blob/d1a67e8b239de6195171bf3ef8cef207e3b8f3a6/pages/api/oauth/authorize.ts#L5

and possibly in other places as well. Look for `github.com` and make changes accordingly.

If you get it working, I wouldn't mind a PR that makes the API host configurable from the `.env` file!

**Comment by syeda-git** [comment]
@laymonage I am now using github.com. https://github.com/syeda-git/webhelp-sandbox
The repo has Giscuss installed, Discussions enabled, and it's public.

![image](https://github.com/giscus/giscus/assets/44337085/c7c2b877-d4fc-4a2f-8ae2-3413d7f930b6)

I have hosted the Giscuss app internally on a VM. When i add syeda-git/webhelp-sandbox in the repository field, it shows "Cannot use giscus on this repository. Make sure all of the above criteria has been met." When i add the same repo on giscuss.app, it's able to connect. Does this mean that my hosted Giscuss is unable to connect to github.com? I can access github.com from the VM.

Syed

**Reply by laymonage** [reply]
I would suggest redoing the same edits as [above](https://github.com/orgs/giscus/discussions/1161#discussioncomment-6842043) and looking into the error response.
